lib/src/model/documentable.dart

```diff
@@ -63,11 +57,16 @@ mixin MarkdownFileDocumentation implements Documentable, Canonicalization {
   String get documentationAsHtml => _documentation.asHtml;
 
   @override
-  bool get hasDocumentation =>
-      documentationFile != null &&
-      packageGraph.resourceProvider
-          .readAsMalformedAllowedStringSync(documentationFile)
-          .isNotEmpty;
+  String get documentation {
+    final docFile = documentationFile;
```

Comment:
I don't think it's possible for documentationFile to change underneath us, but might as well make it explicit.

Reply:
If you meant to cache the `documentationFile`, I've gone ahead and done that as well.

As for the reason why I store it in the local variable is to improve the type inference when we switch to NNBD as it doesn't work for getters.

Reply:
I was meaning that it seemed unlikely the value would change between two calls to the `documentationFile` getter (though admittedly in this mixin, how documentationFile works is not restricted and it's at least theoretically possible).

As for the other bit, I'm a little wary of adding boilerplate just so that type inference works (most of the time, one might as well just add types, if they are needed), but, willing to allow it as it's pretty minor here.